### PR TITLE
Filter using AND if prefix and suffix options are present on S3Agent.

### PR DESCRIPTION
**Pull request by rrodrigu3z**

### CODE REVIEW

File: app/models/agents/s3_agent.rb [hunk 1]
```diff
@@ -234,7 +234,12 @@ def emit_event?(event_type)
 
     def process_file?(file_name)
       return true unless filename_filter_present?
-      prefix_match?(file_name) || suffix_match?(file_name)
+
+      if both_filename_filters_present?
+        prefix_match?(file_name) && suffix_match?(file_name)
```
Comment (duboff) [inline app/models/agents/s3_agent.rb]: Not in love with this approach. Shouldn't both methods only return false if they are present and failing? And then we just always use `&&`?

Comment (rrodrigu3z) [reply app/models/agents/s3_agent.rb]: You're right it was overcomplicated.

File: app/models/agents/s3_agent.rb [hunk 2]
```diff
@@ -233,22 +233,21 @@ def emit_event?(event_type)
...
-      return true unless filename_filter_present?
-      prefix_match?(file_name) || suffix_match?(file_name)
+      prefix_match?(file_name) && suffix_match?(file_name)
     end
 
     def prefix_match?(file_name)
       prefix = interpolated['prefix']
+      return true if prefix.blank?
+
       prefix.present? && file_name.starts_with?(prefix)
```
Comment (duboff) [inline app/models/agents/s3_agent.rb]: Wouldn't the present check here be covered by the blank check above?

Comment (rrodrigu3z) [reply app/models/agents/s3_agent.rb]: 🤦‍♂️ I missed that

**Comment by rrodrigu3z** [comment]
@duboff done.

**Review by duboff — APPROVED** [review]
👍

**Pull request merged**